docs/fundamentals/flow/client.md

```diff
@@ -0,0 +1,314 @@
+(client)=
+# Jina Python Client
+The most convenient way to work with the `Flow` API is the Python Client. It enables you to send `Documents` to the `Flow` API in a number of different ways shown below:
+
+```{admonition} Caution
+:class: caution
+`Flow` provides a `.post()` method that follows the same interface as `client.post()`.
+However, once your solution is deployed in the cloud, the Flow interface is not present anymore.
+Hence, `flow.post()` is not recommended outside of testing or debugging use cases.
+```
+
+```python
+from docarray import Document, DocumentArray
+from jina import Client, Flow
+
+d1 = Document(content='hello')
+d2 = Document(content='world')
+
+
+def doc_gen():
+    for j in range(10):
+        yield Document(content=f'hello {j}')
+
+
+with Flow() as f:
+    client = Client(port=f.port_expose)
+    client.post('/endpoint', d1)  # Single document
+
+    client.post('/endpoint', [d1, d2])  # a list of Document
+
+    client.post('/endpoint', doc_gen)  # Document generator
+
+    client.post('/endpoint', DocumentArray([d1, d2]))  # DocumentArray
+
+    client.post('/endpoint')  # empty
```

Comment:
i suggest us to make it clear. `Flow` is on the server side, client is in the client side, it makes no sense to put client inside of the context manager

Reply:
yes, let's create the `client` outside`

Reply:
makes sense, doing this